miio/integrations/petfeeder/device.py

```diff
@@ -0,0 +1,313 @@
+
+    @command(
+        default_output=format_output(
+            "",
+            lambda result: "".join(
+                (
+                    f"Slot: {plan.slot:02d} "
+                    f"Time: {plan.datetime.astimezone(PetFeeder._local_tz).strftime('%H:%M')} "
+                    f"Units: {plan.units}\n"
+                )
+                for plan in result
+                if plan.enabled
+            ),
+        ),
+    )
+    def get_feed_plans(self) -> List[FeedPlan]:
+        """Get feed plans.
+
+        Return all feeding plans.
+
+        return: List of FeedPlan objects
+        rtype: list
+        raises PetFeederException: if the number of elements returned is wrong
+        """
+        feed_plans = []
+        for i in (1, 2):
+            plans = self.send(f"getfeedplan{i}", [])
+            if len(plans) < 25:
+                raise PetFeederException("Error getting feed plans")
+            for plan in [plans[i : i + 4] for i in range(0, len(plans), 5)]:
+                feed_plans.append(FeedPlan(*plan))
+        return feed_plans
+
+    @command(
+        click.option("-h", "--hour", type=int),
+        click.option("-m", "--minute", type=int),
+        click.option("-u", "--units", type=int),
+        default_output=format_output(
+            "Added feed plan {hour:02d}:{minute:02d} - {units} units"
+        ),
+    )
+    def add_feed_plan(self, hour: int, minute: int, units: int) -> bool:
+        """Add Feed Plan.
+
+        Add a feeding plan.
+        Hours in local tz.
+
+        :param int hour: Hour
+        :param int minute: Minute
+        :param int units: Number of units to dispense (1 unit ~= 5g)
+        :return: MiIO response
+        :rtype: bool
+        :raises PetFeederNoFreeSlots: if there are no unused feed plan slots
+        :raises ValueError: if units < 0 > 30
+        """
+        if units < 0 or units > 30:
+            raise ValueError("Units must be between 1-30")
+        try:
+            slot = next(plan for plan in self.get_feed_plans() if not plan.enabled)
+        except StopIteration:
+            raise PetFeederNoFreeSlots("All feed plan slots in use")
+        plan = replace(slot, hour=hour, minute=minute, units=units, tz=self._local_tz)
+        return self.send("feedListAdd", [plan.format()]) == OK
+
+    @command(
+        click.option("-s", "--slot", type=int, required=True),
+        click.option("-h", "--hour", type=int),
+        click.option("-m", "--minute", type=int),
+        click.option("-u", "--units", type=int),
+        default_output=format_output("Updated feed plan {slot}"),
+    )
+    def edit_feed_plan(self, slot: int, hour: int, minute: int, units: int) -> bool:
+        """Edit Feed Plan.
+
+        Edit a pre-existing feeding plan.
+
+        :param int slot: Feed plan slot
+        :keyword int hour: Hour
+        :keyword int minute: Minute
+        :keyword int units: Number of units to dispense (1 unit ~= 5g)
+        :return: MiIO response
```

Comment:
```suggestion
        :return: True on success
```
I suppose, as this is not returning the response payload?

Reply:
Kinda but it's typed for bool and reStructuredText rtype is set as bool so not really much of an issue.
Technically it is a miio response of yay or nay 😄

Reply:
Honestly, I'd just avoid using any of the rst docstring formatting if not completely necessary. With type hints available, most of the information can be delivered that way already and when not, it might be an indicator that a refactoring is necessary :)

Reply:
More just a force of habit from $dayjob, I see it as additional comments over and above typing.
Whichever docstring format, I think it should be ok to use all available options, rst or google, either works for me.

Some functions really are 1 liner self explanatory, others really do need a full rst docstring to explain what they're doing - and I'm a fan of consistency, if I set rtype on one function then it should be on all.